### DISCUSSION TITLE
A preview command for VS Code extension?

### DISCUSSION THREAD

**Discussion by AustenLamacraft** [quarto-dev/quarto-cli#4275] [Q&A] [answered]
This question has been discussed in [this issue](https://github.com/quarto-dev/quarto-vscode/issues/32) but while the workflow I'm using right now still OK, it could be better...

To get a preview in VS Code with file changes watched, I run the render command in VS Code, which opens a preview but uses the `--no-watch-inputs` flag. I then stop the command at the terminal and run `quarto preview` without `--no-watch-inputs`.

This seems a bit cumbersome. What I'd really like is a preview command, or something, as well as a render command, which runs
 `quarto preview` without `--no-watch-inputs`. Or maybe a setting in the extension settings to change the default behaviour

**Comment by mcanouil** [comment, marked as answer]
You can turn on the "render on save" feature `quarto.render.renderOnSave` (disabled by default for performance reason).

<img width="1161" alt="image" src="https://user-images.githubusercontent.com/8896044/217517561-650cba82-98d4-46a2-9e6a-c2691b71ec33.png">

**Reply by AustenLamacraft** [reply]
Ah thanks, that's great!

**Comment by astrowonk** [comment]
The "render on save" feature is good, but why does the `render` command in the command palette default to `--no-watch-inputs` with no way to change that? It's always confused me. The text in the terminal even says "Watching files for changes" but it's not.

**Reply by mcanouil** [reply]
See:

- https://github.com/quarto-dev/quarto-cli/discussions/1939#discussioncomment-3417704
